search-resources-api/src/test/java/no/unit/nva/search2/SwsResourceHandlerTest.java

```diff
@@ -0,0 +1,317 @@
+    }
+
+    @Test
+    void shouldReturnSortedSearchResultsWhenSendingContributorId() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getInputStreamWithContributorId(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+    }
+
+    @Test
+    void shouldSearchResultsWhenSendingContributorIdAndBadResponseFromPreferencesApi()
+        throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getInputStreamWithContributorId(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+    }
+
+    @Test
+    void shouldNotReturnSortedSearchResultsWhenSendingMultipleContributorId()
+        throws IOException {
+        prepareRestHighLevelClientOkResponse();
+
+        handler.handleRequest(getInputStreamWithMultipleContributorId(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+    }
+
+    @Test
+    @Disabled
+    void shouldReturnAggregationAsPartOfResponseWhenDoingASearch() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+
+        handler.handleRequest(getInputStream(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+        var expected = getSearchResourcesResponseFromFile(ROUNDTRIP_RESPONSE_JSON);
+        var actualBody = gatewayResponse.body();
+
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+        assertThat(actualBody, is(equalTo(expected)));
+        assertNotNull(actualBody.aggregations());
+        assertNotNull(actualBody.aggregations().get("Bidragsyter"));
+    }
+
+    @Test
+    void shouldReturnSearchResultsWithEmptyHitsWhenQueryResultIsEmpty() throws IOException {
+        prepareRestHighLevelClientEmptyResponse();
+
+        handler.handleRequest(getInputStream(), outputStream, mock(Context.class));
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+        var actualBody = gatewayResponse.body();
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+        assertThat(actualBody.totalHits(), is(equalTo(0L)));
+        assertThat(actualBody.hits(), is(empty()));
+        assertDoesNotThrow(() -> gatewayResponse.body().id().normalize());
+    }
+
+    @Test
+    void shouldReturn200WhenSortOrderIsNotSpecified() throws IOException {
+        prepareRestHighLevelClientEmptyResponseForSortOrder("desc");
```

Comment:
is it specified here or is is some default param? test says "sort order not specified"